sdk/storage/azure-storage-blob-changefeed/src/main/java/com/azure/storage/blob/changefeed/BlobLazyDownloader.java

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob.changefeed;
+
+import com.azure.storage.blob.BlobAsyncClient;
+import com.azure.storage.blob.implementation.util.ChunkedDownloadUtils;
+import com.azure.storage.blob.models.BlobDownloadAsyncResponse;
+import com.azure.storage.blob.models.BlobRange;
+import com.azure.storage.blob.models.BlobRequestConditions;
+import com.azure.storage.blob.models.ParallelTransferOptions;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+import java.nio.ByteBuffer;
+import java.util.function.Function;
+
+/**
+ * FOR INTERNAL USE ONLY.
+ * Class to lazily download a blob.
+ */
+class BlobLazyDownloader {
+
+    private final BlobAsyncClient client; /* Client to download from. */
+    private final long blockSize; /* The block size. */
+    private final BlobRange range;
+
+    /**
+     * Creates a new BlobLazyDownloader to download the rest of a blob at a certain offset.
+     */
+    BlobLazyDownloader(BlobAsyncClient client, long blockSize, long offset) {
+        this.client = client;
+        this.blockSize = blockSize;
+        this.range = new BlobRange(offset);
+    }
+
+    /**
+     * Creates a new BlobLazyDownloader to download a partial blob.
+     */
+    BlobLazyDownloader(BlobAsyncClient client, long totalSize) {
+        this.client = client;
+        this.blockSize = totalSize;
+        this.range = new BlobRange(0, totalSize);
+    }
+
+    /* TODO (gapra) : It may be possible to unduplicate the code below as well to share between downloadToFile but
+       wasnt immediately obvious to me */
+    public Flux<ByteBuffer> download() {
+        ParallelTransferOptions options = new ParallelTransferOptions()
+            .setBlockSizeLong(blockSize);
+        BlobRequestConditions requestConditions = new BlobRequestConditions();
+
+        Function<BlobRange, Mono<BlobDownloadAsyncResponse>> downloadFunc = range
+            -> client.downloadWithResponse(range, null, new BlobRequestConditions(), false);
+
+        return ChunkedDownloadUtils.downloadFirstChunk(range, options, requestConditions, downloadFunc,
+            Schedulers.immediate())
```

Comment:
Why specify the scheduler here?

Reply:
It looks like we specified the scheduler in the original download to file code that you refactored. Weird. I don't recall why we would've done that.

In that case, why use immediate here instead of elastic?

Reply:
Changing to elastic and I'll remove the Schedulers method param